parse/parse.go

```diff
@@ -636,6 +636,11 @@ func (p *parser) parseMatcher() (Matcher, error) {
 
 selectorsLoop:
 	for {
+		itm := p.next()
+		if itm.typ != itemWhitespace {
+			return errorf("%w", unexpectedErr(itm, itemWhitespace))
+		}
+
```

Comment:
when lexer lexes whitespace, it concatenates all whitespace characters and emits whitespace item. Effectively, the lexer won't emit two sequential whitespace items.  Please use `next()` instead of `nextNonWS()`!